qiita_ware/test/test_ebi.py

```diff
@@ -405,6 +423,78 @@ def test_generate_curl_command(self):
                          '%20ebi_access_key%3D"')
         self.assertEqual(obs, exp_with_cert)
 
+    def test_generate_demultiplexed_fastq(self):
+        # generating demux file for testing
+        if not isdir(self.temp_dir):
+            makedirs(self.temp_dir)
+
+        fna_fp = join(self.temp_dir, 'seqs.fna')
+        self.files_to_remove.append(fna_fp)
+        # demux_fp doesn't need to be removed cause is being moved to be
+        # the ppd being created and will be removed there
+        demux_fp = join(self.temp_dir, 'demux.seqs')
+        with open(fna_fp, 'w') as f:
+            f.write(FASTA_EXAMPLE)
+        with File(demux_fp, "w") as f:
+            to_hdf5(fna_fp, f)
+
+        # Adding a new PreprocessedData to study 1
+        # The 3rd parameter 1 is the PreprocessedIlluminaParams id, which is
+        # not relevant for this test
+        exp_demux_samples = set(
+            ['1.SKD6.640190', '1.SKM6.640187', '1.SKD9.640182',
+             '1.SKB2.640194', '1.SKM8.640201', '1.SKM4.640180',
+             '1.SKM2.640199', '1.SKB3.640195', '1.SKB6.640176'])
+        ppd = PreprocessedData.create(Study(1),
+                                      "preprocessed_sequence_illumina_params",
+                                      1, [(demux_fp, 6)],
+                                      prep_template=PrepTemplate(1))
+        self.ppd_to_remove.append(ppd.id)
+        [self.files_to_remove.append(fps) for _, fps, _ in ppd.get_filepaths()]
```

Comment:
I think this construction is really strange, you're defining a list of None that you're never using...

Can you use this instead?
```python
self.files_to_remove.extend([fps for _, fps, _ in ppd.get_filepaths()])
```